dart/lib/src/metrics/metrics_api.dart

```diff
@@ -12,14 +15,80 @@ class MetricsApi {
       {final double value = 1.0,
       final SentryMeasurementUnit? unit,
       final Map<String, String>? tags}) {
-    _hub.metricsAggregator?.increment(
+    _hub.metricsAggregator?.emit(
+      MetricType.counter,
       key,
       value,
       unit ?? SentryMeasurementUnit.none,
       _enrichWithDefaultTags(tags),
     );
   }
 
+  /// Emits a Gauge metric, identified by [key], adding [value] to it.
+  /// You can set the [unit] and the optional [tags] to associate to the metric.
+  void gauge(final String key,
+      {required final double value,
+      final SentryMeasurementUnit? unit,
+      final Map<String, String>? tags}) {
+    _hub.metricsAggregator?.emit(
+      MetricType.gauge,
+      key,
+      value,
+      unit ?? SentryMeasurementUnit.none,
+      _enrichWithDefaultTags(tags),
+    );
+  }
+
+  /// Emits a Gauge metric, identified by [key], adding [value] to it.
+  /// You can set the [unit] and the optional [tags] to associate to the metric.
+  void distribution(final String key,
+      {required final double value,
+      final SentryMeasurementUnit? unit,
+      final Map<String, String>? tags}) {
+    _hub.metricsAggregator?.emit(
+      MetricType.distribution,
+      key,
+      value,
+      unit ?? SentryMeasurementUnit.none,
+      _enrichWithDefaultTags(tags),
+    );
+  }
+
+  /// Emits a Set metric, identified by [key], adding [value] or the CRC32
+  ///  checksum of [stringValue] to it.
+  /// Providing both [value] and [stringValue] adds both values to the metric.
+  /// You can set the [unit] and the optional [tags] to associate to the metric.
+  void set(final String key,
+      {final int? value,
+      final String? stringValue,
+      final SentryMeasurementUnit? unit,
+      final Map<String, String>? tags}) {
+    if (value != null) {
+      _hub.metricsAggregator?.emit(
+        MetricType.set,
+        key,
+        value,
+        unit ?? SentryMeasurementUnit.none,
+        _enrichWithDefaultTags(tags),
+      );
+    }
+    if (stringValue != null && stringValue.isNotEmpty) {
+      final intValue = Crc32Utils.getCrc32(utf8.encode(stringValue));
+
+      _hub.metricsAggregator?.emit(
+        MetricType.set,
+        key,
+        intValue,
+        unit ?? SentryMeasurementUnit.none,
+        _enrichWithDefaultTags(tags),
+      );
+    }
+    if (value == null && (stringValue == null || stringValue.isEmpty)) {
+      _hub.options.logger(
+          SentryLevel.info, 'No value provided. No metric will be emitted.');
```

Comment:
This can happen because the metrics are flushed at specific intervals, but are retained for a different amount of time. This is not a fault of the user, and not an error in general.
E.g. Every 5 seconds we flush metrics, but the metrics are retained in buckets of 10 seconds.